torch/linalg/__init__.py

```diff
@@ -184,3 +184,94 @@
     >>> torch.allclose(torch.tensordot(a, x, dims=x.ndim), b, atol=1e-6)
     True
 """)
+
+
+qr = _add_docstr(_linalg.linalg_qr, r"""
+qr(input, mode='reduced', *, out=None) -> (Tensor, Tensor)
+
+Computes the QR decomposition of a matrix or a batch of matrices :attr:`input`,
+and returns a namedtuple (Q, R) of tensors such that :math:`\text{input} = Q R`
+with :math:`Q` being an orthogonal matrix or batch of orthogonal matrices and
+:math:`R` being an upper triangular matrix or batch of upper triangular matrices.
+
+Depending on the value of :attr:`mode` this function returns the reduced or
+complete QR factorization. See below for a list of valid modes.
+
+.. warning:: **Differences with** :meth:`~torch.qr`:
+
+             * ``torch.qr`` takes a boolean parameter ``some`` instead of ``mode``:
+
+               - ``some=True`` is equivalent of ``mode='reduced'``: both are the
+                 default
+
+               - ``some=False`` is equivalent of ``mode='complete'``.
+
+             **Differences with** ``numpy.linalg.qr``:
```

Comment:
For the differences with NumPy, instead of a warning we can probably have a note like:

"Unlike NumPy's linalg.qr, this function always returns a tuple of two tensors. When \`\`mode='r'\`\`, the \`Q\` tensor is an empty tensor."